pkg/controller/backup_sync_controller.go

```diff
@@ -107,4 +126,33 @@ func (c *backupSyncController) run() {
 			}
 		}
 	}
+
+	deleteUnused(c, cloudBackupNames)
+	return
+}
+
+// deleteUnused deletes backups from Kubernetes if they do not exist in object storage
+func deleteUnused(c *backupSyncController, cloudBackupNames sets.String) {
```

Comment:
I would make this a method on `backupSyncController` instead of passing it in as a parameter